android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -103,29 +103,47 @@ open class PolyOut(
 
     open suspend fun readdir(
         id: String
-    ): Array<String> {
-
+    ): Array<Map<String, String>> {
         val fs = Preferences.getFileSystem(context)
+        val retList = mutableListOf(mapOf<String,String>())
+        
+        retList.removeFirst()
         if (id == "") {
             val newFs = fs.filter {
                 File(idToPath(it.key, context)).exists()
             }
             Preferences.setFileSystem(context, newFs)
-            return newFs.keys.toTypedArray()
+
+            val allIds = newFs.keys;
+            
+            for (idPath in allIds){
+                val relPath = getRelativePathFromId(idPath)
+                val idMap = mapOf<String,String>("id" to idPath, "path" to relPath)
+                retList.add(idMap)
```

Comment:
Suppose this could all be shortened, instead of the old `return newFs.keys.toTypedArray()` you could just do:

```
return newFs.keys.map {
    mapOf<String, String>("id" to it, "path" to it)
}.toTypedArray();
```